src/mono/mono/mini/interp/transform.c

```diff
@@ -10188,61 +10171,96 @@ add_active_call (TransformData *td, ActiveCalls *ac, InterpInst *call)
 		return;
 
 	if (ac->active_calls_count == ac->active_calls_capacity) {
-		ActiveCall *old = ac->active_calls;
+		InterpInst **old = ac->active_calls;
 		ac->active_calls_capacity *= 2;
-		ac->active_calls = (ActiveCall*)mono_mempool_alloc (td->mempool, ac->active_calls_capacity * sizeof (ActiveCall));
-		memcpy (ac->active_calls, old, ac->active_calls_count * sizeof (ActiveCall));
+		ac->active_calls = (InterpInst**)mono_mempool_alloc (td->mempool, ac->active_calls_capacity * sizeof (InterpInst*));
+		memcpy (ac->active_calls, old, ac->active_calls_count * sizeof (InterpInst*));
 	}
-
-	ac->active_calls [ac->active_calls_count].call = call;
-	ac->active_calls [ac->active_calls_count].param_size = get_call_param_size (td, call);
-	ac->param_size += ac->active_calls [ac->active_calls_count].param_size;
+	ac->active_calls [ac->active_calls_count] = call;
 	ac->active_calls_count++;
 
 	// Mark a flag on it so we don't have to lookup the array with every argument store.
 	call->flags |= INTERP_INST_FLAG_ACTIVE_CALL;
 }
 
+/**
+ * Function allocates offsets of resolved calls following a constraint
+ * where the base offset of a call must be greater than the offset of any argument of other active call args.
+ *
+ * Function first removes the call from an array of active calls. If a match is found,
+ * the call is removed from the array by moving the last entry into its place. Otherwise, it is a call without arguments.
+ *
+ * Call is push onto the stack with an array of other active calls that need to be resolved first.
```

Comment:
I think this comment is outdated. You might want to double check the explanation for this method.

Reply:
Updated.